src/main/java/org/example/dao/RouteDAOImpl.java

```diff
@@ -0,0 +1,108 @@
+package org.example.dao;
+
+import org.example.configurations.ConnectionPool;
+import org.example.model.Point;
+import org.example.model.Route;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class RouteDAOImpl implements RouteDAO {
+    private static final ConnectionPool CONNECTIONPOOL = ConnectionPool.getInstance();
+
+    public RouteDAOImpl() {
+    }
+    @Override
+    public void insertRoute(Route route) {
+        try (Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+             PreparedStatement statement = connection.prepareStatement("INSERT INTO routes (startPointId, endPointId, shortestDistance) VALUES (?, ?, ?)", Statement.RETURN_GENERATED_KEYS)) {
```

Comment:
This should be
INSERT INTO routes (start_point, end_point, distance) VALUES (?, ?, ?)